Run New Test Module

wanfei-imda opened the pull request:
## Description

[Provide a brief description of the changes or features introduced by this pull request.]

## Motivation and Context

1. to run new test on portal
2. export and import checklists

## Type of Change

## How to Test
1. start the portal, and start the test engine worker
2. navigate to /models/upload/, and upload sample data file sample_bc_credit_data.sav
3. navigate to /datasets/upload/, and upload sample model file sample_bc_credit_sklearn_linear.LogisticRegression.sav
4. navigate to /results/run:
if server is inactive, there will be a modal popup stating server is inactive.
if server is active:
- select aiverfy_accumulated_local_effect from the algorithm dropdown
- select the model file and data file uploaded previously
- click submit

upon submitting, it will redirect to /results/run/view_tests/, where there is a list of all tests.

export and import checklists
1. navigate to /inputs/, and click into process checklists, and click on add input and create new 
2. fill in some of the questions
3. page redirected to listing page of checklists -> click on existing group -> click on download button in group name header. check the excel to see it is correctly filled in and formatted. edit 1 column e.g. add a Yes.
4. navigate to add input again and click upload excel -> try uploading the excel you just downloaded. check there is a modal popup to ask if you want to overwrite. select yes. check it is successful in overwriting.


## Checklist

Please check all the boxes that apply to this pull request using "x":

- [ ]  I have tested the changes locally and verified that they work as expected.
- [ ]  I have added or updated the necessary documentation (README, API docs, etc.).
- [ ]  I have added appropriate unit tests or functional tests for the changes made.
- [ ]  I have followed the project's coding conventions and style guidelines.
- [ ]  I have rebased my branch onto the latest commit of the main branch.
- [ ]  I have squashed or reorganized my commits into logical units.
- [ ]  I have added any necessary dependencies or packages to the project's build configuration.
- [ ]  I have performed a self-review of my own code.
- [ ]  I have read, understood and agree to the Developer Certificate of Origin below, which this project utilises.

## Screenshots (if applicable)

[If the changes involve visual modifications, include screenshots or GIFs that demonstrate the changes.]

## Additional Notes

[Add any additional information or context that might be relevant to reviewers.]

<details>
  <summary>Developer Certificate of Origin</summary>

 ```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
 ```
</details>

imda-peckyoke commented:
Encounter following error when I run the portal
![image](https://github.com/user-attachments/assets/e9192b61-3746-4efd-b864-5399f85008a2)

imda-benedictlee commented:
Docker Setup:

Using Docker APIGW, PORTAL & TEST ENGINE:

**Test Runs**
<img width="1508" alt="Screenshot 2025-04-04 at 5 14 33 PM" src="https://github.com/user-attachments/assets/1677e2bf-3684-4635-a9f9-c97c024f2577" />

**Checklist**
<img width="1259" alt="Screenshot 2025-04-04 at 5 47 25 PM" src="https://github.com/user-attachments/assets/0fe1f611-6aff-40bf-b7dd-4eb0f5207429" />

**Check List 1 Excel Sheet**
[Checklist 1_checklists.xlsx](https://github.com/user-attachments/files/19601795/Checklist.1_checklists.xlsx)

**Check List 2 Excel Sheet**
[Checklist 2_checklists.xlsx](https://github.com/user-attachments/files/19601806/Checklist.2_checklists.xlsx)

**Imported Check List 2**
<img width="1249" alt="Screenshot 2025-04-04 at 5 47 38 PM" src="https://github.com/user-attachments/assets/18599885-f576-4b25-a330-3a387b91d171" />

imda-benedictlee commented:
Additional Issues to Link:
- #484

wanfei-imda commented:
1. Fixed bugs:
- [x] Summary Justification will be updated when uploaded when updated excel is upload to portal
- [x] Added validation for Complete column to be Yes, No or Not Applicable values only.

2. Updated description on Excel Upload page for clearer instructions.
2. Added Export JSON

imda-benedictlee reviewed (approved):
LGTM. All Defects Fixed and Retested.
